plugins/UltimakerMachineActions/UMOCheckupMachineAction.qml

```diff
@@ -0,0 +1,271 @@
+import UM 1.2 as UM
+import Cura 1.0 as Cura
+
+import QtQuick 2.2
+import QtQuick.Controls 1.1
+import QtQuick.Layouts 1.1
+import QtQuick.Window 2.1
+
+Cura.MachineAction
+{
+    anchors.fill: parent;
+    Item
+    {
+        id: checkupMachineAction
+        anchors.fill: parent;
+        property int leftRow: checkupMachineAction.width * 0.40
+        property int rightRow: checkupMachineAction.width * 0.60
+        UM.I18nCatalog { id: catalog; name:"cura"}
+        Label
+        {
+            id: pageTitle
+            width: parent.width
+            text: catalog.i18nc("@title", "Check Printer")
+            wrapMode: Text.WordWrap
+            font.pointSize: 18;
+        }
+
+        Label
+        {
+            id: pageDescription
+            anchors.top: pageTitle.bottom
+            anchors.topMargin: UM.Theme.getSize("default_margin").height
+            width: parent.width
+            wrapMode: Text.WordWrap
+            text: catalog.i18nc("@label","It's a good idea to do a few sanity checks on your Ultimaker. You can skip this step if you know your machine is functional");
+        }
+
+        Item
+        {
+            id: startStopButtons
+            anchors.top: pageDescription.bottom
+            anchors.topMargin: UM.Theme.getSize("default_margin").height
+            anchors.horizontalCenter: parent.horizontalCenter
+            height: childrenRect.height
+            width: startCheckButton.width + skipCheckButton.width + UM.Theme.getSize("default_margin").height < checkupMachineAction.width ? startCheckButton.width + skipCheckButton.width + UM.Theme.getSize("default_margin").height : checkupMachineAction.width
+            Button
+            {
+                id: startCheckButton
+                anchors.top: parent.top
+                anchors.left: parent.left
+                text: catalog.i18nc("@action:button","Start Printer Check");
+                onClicked:
+                {
+                    checkupContent.visible = true
+                    startCheckButton.enabled = false
+                    manager.startCheck()
+                }
+            }
+
+            Button
+            {
+                id: skipCheckButton
+                anchors.top: parent.width < checkupMachineAction.width ? parent.top : startCheckButton.bottom
+                anchors.topMargin: parent.width < checkupMachineAction.width ? 0 : UM.Theme.getSize("default_margin").height/2
+                anchors.left: parent.width < checkupMachineAction.width ? startCheckButton.right : parent.left
+                anchors.leftMargin: parent.width < checkupMachineAction.width ? UM.Theme.getSize("default_margin").width : 0
+                text: catalog.i18nc("@action:button", "Skip Printer Check");
+                onClicked: manager.setFinished()
+            }
+        }
+
+        Item
```

Comment:
This would benefit from being in a Grid instead of an Item; you would not have to do all the anchoring